kore/src/Kore/Exec.hs

```diff
@@ -458,17 +458,16 @@ rpcExec
 
 -- | Modify a 'TransitionRule' to track the depth of the execution graph.
 trackExecDepth ::
-    TransitionRule monad rule state ->
-    TransitionRule monad rule (ExecDepth, state)
+    TransitionRule monad rule (ProgramState p) ->
+    TransitionRule monad rule (ExecDepth, ProgramState p)
 trackExecDepth transit prim (execDepth, execState) = do
     execState' <- transit prim execState
     let execDepth' = (if didRewrite execState' then succ else id) execDepth
     pure (execDepth', execState')
   where
-    didRewrite _ = isRewrite prim
-
-    isRewrite Rewrite = True
-    isRewrite _ = False
+    didRewrite Kore.Rewrite.Bottom = prim == Rewrite
```

Comment:
this seems like the meat of the issue. could you add an in code comment what this is checking?